### PR TITLE
Tests For Querier and Postings

### PR DESCRIPTION
**Pull request by Gouthamve**
Tests for Querier and Postings.

### CODE REVIEW

File: querier.go [hunk 1]
```diff
@@ -466,8 +466,8 @@ func (s *chunkSeries) Iterator() SeriesIterator {
 // SeriesIterator iterates over the data of a time series.
 type SeriesIterator interface {
 	// Seek advances the iterator forward to the given timestamp.
-	// If there's no value exactly at ts, it advances to the last value
-	// before tt.
+	// If there's no value exactly at t, it advances to the first value
```
Comment (Gouthamve) [inline querier.go]: Fixes #37

**Comment by Gouthamve** [comment]
The commit `b60c206` fixes #43 and I can open a separate PR for it if required.

File: querier_test.go [hunk 2]
```diff
@@ -200,3 +203,641 @@ func expandSeriesIterator(it SeriesIterator) (r []sample, err error) {
...
+
+	cases := []struct {
+		dataIdx int
+
+		mint, maxt int64
+		ms         []labels.Matcher
+		exp        SeriesSet
+	}{
+		{
+			dataIdx: 0,
```
Comment (fabxc) [inline querier_test.go]: No need to complicate things for multiple datasets if we are not using it yet. We never might.

File: index_test.go [hunk 3]
```diff
@@ -306,6 +314,24 @@ func TestPersistence_index_e2e(t *testing.T) {
...
+			require.NoError(t, err)
+
+			strsRes, err := tplsRes.At(i)
+			require.NoError(t, err)
+
+			require.Equal(t, strsExp, strsRes)
+		}
+	}
+
+	require.NoError(t, ir.Close())
```
Comment (Gouthamve) [inline index_test.go]: Hmm, removed index.go from the checklist of done things. Will test index once I get some free time. But I don't think this should block this PR.

**Comment by Gouthamve** [comment]
Incorporated the feedback, PTAL. 

Also I am caught up right now and will resume adding tests after next week, so it would be great if this could be merged as is.

**Comment by fabxc** [comment]
👍

**Pull request merged**